test/resolver/missing_main/package.json

```diff
@@ -0,0 +1,3 @@
+{
+  "main": "index.js"
+}
```

Comment:
it might be nice to add *another* fixture for when "main" points to "not index.js", and that main is missing, and "index.js" is also missing :-)

Reply:
Yep. On this.

Reply:
@ljharb Added relevant test cases. Ready for review.